.gitlab-ci.yml

```diff
@@ -1,43 +1,48 @@
-image: docker:stable
+stages:
+  - lint
+  - test
+  - coverage
+  - deploy
 
-services:
-  - docker:dind
+kick-lint:
+  stage: lint
+  image: rust:1.30
+  allow_failure: true
+  script:
+    - rustup component add rustfmt-preview
+    - cargo fmt --all -- --check
+    - rustup component add clippy-preview
+    - cargo clippy -- --allow "clippy::redundant_field_names"
 
-before_script:
-  - apk --no-cache add curl jq
-  
-kick-e2e:
-    tags:
-        - docker
-    stage: deploy
-    script:
-      - set -e
+kick-test:
+  stage: test
+  image: rust:1.30
+  script:
+    - cargo test --verbose
 
-      # kick e2e
-      - >-
-        E2E_PIPELINE_ID=`curl
```

Comment:
Thanks @ceseale 

It would be great to have you give a more thorough review the rest of this. I was kindof skeptical of the deploy task in genera, but put it back in because of git history that (and I'm paraphrasing) "this should be this way because the rest doesn't do anything".

Reply:
Yeah, that's an old comment because at the time relay wasn't doing anything. For this project to be deployed it has to run in E2E.